src/datachain/query/dataset.py

```diff
@@ -396,7 +397,9 @@ def process_udf_outputs(
         for row in udf_output:
             cb.relative_update()
             rows.append(adjust_outputs(warehouse, row, udf_col_types))
-            if len(rows) >= batch_size:
+            if len(rows) >= batch_size or (
+                len(rows) % 10 == 0 and psutil.virtual_memory().percent > 80
+            ):
```

Comment:
@shcheklein @dreadatour Thoughts on this heuristic as a simple way to automatically check for memory limits instead of having to set them manually? For comparison, dask checks every 200 ms and pauses workers when they reach 80% of the memory limit.

Reply:
sounds good to me. We should try this.

Reply:
Wow, sounds good! Love the idea and simplicity.